### ISSUE TITLE
Predator mutations don't get a mood boost for human flesh

### ISSUE THREAD

**Issue by KittyTac** [CleverRaven/Cataclysm-DDA#49238]
# Describe the bug

If you eat human flesh you don't get the oddly-named "Ate Vegetables" boost that predators get when eating animal flesh.

# Steps To Reproduce

1. Debug in Predator/Apex Predator
2. Spawn a NPC.
3. Kill NPC.
4. Butcher NPC's corpse.
5. Eat NPC's meat.

# Expected behavior

Predators don't care about eating humans any more or less than other animals. Long pork is just meat to them, as evidenced by Sapiovore.

# Screenshots

Not needed.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: a40d493 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Bionic Professions [package_bionic_professions],
    Kittymod [kittymod],
    Tankmod: Revived [Tankmod_Revived],
    Tankmod: Revived - UndeadPeople Tileset [Tankmod_RevivedUDP],
    No Fungal Monsters [No_Fungi]
]

# Additional context

Kittymod is my personal mod, doesn't change those mutations. As I understand this it's hardcoded anyways. For the record, they do not get the cannibalism penalty so maybe this is intended.

**Comment by actual-nh** [comment]
This may well be true of demihuman meat as well, but I don't know - work to do...

**Comment by leemuar** [comment]
human flesh has PREDATOR_FUN flag
And there is an interesting comment [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/4b40b34725c665c7fc6a7dd841a70524dc4b0cc4/src/consumption.cpp#L1185)^

> // The PREDATOR_FUN flag shouldn't be on human flesh, to not interfere with sapiovores/cannibalism.

**Comment by KittyTac** [comment]
Well predators eating meat gets you the "Ate Vegetables" bonus, you can test it. I thought it was just something hacky so I didn't report it. Guess that's another bug, I think they should be swapped around.

**Comment by leemuar** [comment]
Yeah, looks like a hack, "Ate Vegetables" was probably intended to be a negative morale effect, not positive. I'll open another issue on that

> 
> 
> Well predators eating meat gets you the "Ate Vegetables" bonus, you can test it. I thought it was just something hacky so I didn't report it. Guess that's another bug, I think they should be swapped around. But fair enough on that part about it not interfering.

**Comment by leemuar** [comment]
@hjk321 could you please review this issue as original author of predator_fun flag?

**Comment by hjk321** [comment]
Yeah, whoops, I didn't really know what I was doing. It should be swapped. The "Ate Veggies" message was an unintended side effect.

**Comment by Night-Pryanik** [comment]
Fixed by #49861.

**Comment by leemuar** [comment]
> 
> 
> Fixed by #49861.

@Night-Pryanik this issue should be reopened because #49861 does not solves this. Current issue is about human flesh not giving morale boost for predators. #49861 only fixes morale boost description.

**Comment by hjk321** [comment]
> Predators don't care about eating humans any more or less than other animals. Long pork is just meat to them, as evidenced by Sapiovore.

Sapiovore is a direct upgrade from those two mutations. Giving them the benefits of sapivore would not make any sense whatsoever. As far as I am aware this is intended behavior. If you want the boost without taking Sapiovore, which I should remind you is a direct upgrade from the rest of the predator mutations, then take cannibalism on top of it. 

Predator isn't Sapiovore. Apex Predator isn't Sapiovore. Sapiovore is Sapiovore. The system is working as intended, this issue should be closed.

**Comment by Stadler76** [comment]
See [this](https://github.com/CleverRaven/Cataclysm-DDA/blob/57fd5acf90b6f2930e5d9655b5d3ac8d44643240/src/consumption.cpp#L1183-L1224):
According to this comment:
```cpp
// Sapiovores don't recognize humans as the same species.
// But let them possibly feel cool about eating sapient stuff - treat like psycho
```

Shouldn't this chunk:
```cpp
} else if( cannibal && psycho ) {
    add_msg_if_player( m_good, _( "You feast upon the human flesh." ) );
    add_morale( MORALE_CANNIBAL, 15, 200 );
} else if( ( cannibal || sapiovore ) && spiritual ) {
```

instead read:
```cpp
} else if( ( cannibal || sapiovore ) && psycho ) {
    add_msg_if_player( m_good, _( "You feast upon the human flesh." ) );
    add_morale( MORALE_CANNIBAL, 15, 200 );
} else if( ( cannibal || sapiovore ) && spiritual ) {
```
?

**Comment by hjk321** [comment]
Yes, that looks like a good idea. I can't quite remember whether there's already a morale boost message printing for sapiovore though.

**Comment by Stadler76** [comment]
> Yes, that looks like a good idea. I can't quite remember whether there's already a morale boost message printing for sapiovore though.

Yes, there is:
```cpp
} else if( psycho || sapiovore ) {
    add_msg_if_player( _( "Meh.  You've eaten worse." ) );
} else if( spiritual ) {
```

btw: In [`consumption.cpp`](https://github.com/CleverRaven/Cataclysm-DDA/blob/57fd5acf90b6f2930e5d9655b5d3ac8d44643240/src/consumption.cpp#L1207) line 1207 could be reduced to:
```cpp
} else if( psycho && spiritual ) {
```
since sapoivore && spritual is already handled in line 1200

PS: I could write a PR for that, but I'm currently struggling with running the unit-tests so it would be neat, if someone who has more experience with the test cases could file the PR for it.
